### ISSUE TITLE
Modify Error Boundaries documetation

### ISSUE THREAD

**Issue by yekver** [reactjs/reactjs.org#145]
It'll be cool to mention in documentation that error boundaries do not work with:
 - Server Side Rendering (see: https://github.com/facebook/react/issues/10442)
 - Event Handlers
 - Async code in render

Also it may be helpful to provide some code how to deal with event handlers, for example:
```javascript
class MyComponent extends React.Component {
  state = { err: null }
  
  handleClick = () => {
    try {
      // something that produces an error
    } catch(err) {
      this.setState({ err })
    }
  }

  render() {
    if (this.state.err) {
        throw this.state.err
    }
    // ...
  }
}
```

**Comment by bvaughn** [comment]
Hm. Interesting feedback.

The docs _do_ hint at the fact that event handlers and async code aren't covered:
>  Error boundaries catch errors during **rendering, in lifecycle methods, and in constructors**

But I agree that we could be more explicit, particularly about the SSR case. 👍

**Comment by thomasyimgit** [comment]
@bvaughn I can look into this. So we just explicitly tell the users that these cases are not covered in the `Note` part? Should we add a similar code example as @yekver posted?

**Comment by bvaughn** [comment]
Thanks @thomasyimgit! This issue is all yours! 😄

I think a note _and_ examples both would be helpful. This is an "advanced" docs page after all so it's okay to go into a bit more detail I think. Use your judgement and we'll be happy to give you feedback!

I've added an "in-progress" label so that others will know not to start work on the issue. If you change your mind about the issue, no worries! Just let me know so that I can remove the label and free it up for someone else to claim.

Cheers!

**Comment by bvaughn** [comment]
Resolved by PR #175
